controller/src/main/java/io/pravega/controller/eventProcessor/impl/EventProcessorCell.java

```diff
@@ -146,6 +152,15 @@ protected final void shutDown() throws Exception {
             }
         }
 
+        @Override
+        protected void triggerShutdown() {
+            log.info("Event processor triggerShutdown called for {}", objectId);
+            if (this.currentThread != null && Thread.State.WAITING.equals(this.currentThread.getState())) {
```

Comment:
Regardless of the state it should interrupt.

Reply:
Handled the condition check in the stopAsync method's parameter of EventProcessorCell class.